src/app/core/component/graph-display/graph-display.component.ts

```diff
@@ -210,18 +255,35 @@ export class GraphDisplayComponent implements AfterContentInit, OnChanges {
   }
 
   nodeClicked(event) {
-    const node = event['detail']['node'];
-    if (event['detail']['key'] === 'info') {
-      this.router.navigate([UriConverter.doubleEncodeUri(node.uri)], {relativeTo: this.activatedRoute});
+    const node = event[ 'detail' ][ 'node' ];
+    if (event[ 'detail' ][ 'key' ] === 'info') {
+      this.router.navigate([UriConverter.doubleEncodeUri(node.uri)], { relativeTo: this.activatedRoute });
+      return;
+    }
+    if (event[ 'detail' ][ 'key' ] === 'delete') {
+      this.deletePatternEvent.emit(node.id);
+      return;
     }
     this.showInfoForClickedNode(node);
   }
 
+  nodePositionChanged(event) {
```

Comment:
A function does something, i.e., a name should indicate this. Although this is an event listener it is much more readable of the name indicates what's going on: e.g., handleNodePositionChangedEvent might be better.

Reply:
Done